test/MUIDataTable.test.js

```diff
@@ -211,6 +220,67 @@ describe('<MUIDataTable />', function() {
     assert.deepEqual(state.data, expectedResult);
   });
 
+  it('should correctly re-build internal table data and displayData structure with columns prop change', () => {
```

Comment:
Much appreciate the attempt to test the library and the changes, however, both of these tests pass in `master` or with all your other changes here removed, meaning that they don't actually demonstrate that your code fixes the problem. This is part of the difficulty with logic in the lifecycle methods, in that it is very difficult to mimic their behavior in the same way as in actual usage of the library.